### PR TITLE
Add shell completion for get/completion

### LINKED ISSUE #158: add shell completion
Feature request to have additional command to generate shell completion.

```shell
arkade completion
``` 

I can take this task if that's ok

### PR DESCRIPTION
**Pull request by tuananh**
Signed-off-by: Tuan Anh Tran <me@tuananh.org>

Add shell completion for bash and zsh

I've only add completion for `get` command. for other commands (info, install and update`, we will have to maintain a list of apps. If you want, i can add it.

## Description

Add shell completion for bash and zsh

## Motivation and Context

Attempt to close #158 

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Test with bash on ubuntu

image
image

running `arkade completion bash` to generate completion for bash

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by Waterdrips** [comment]
This gave me the idea for this PR: https://github.com/alexellis/arkade/pull/161/files

It should (if merged?) give us the ability to get a list of the apps for `install` by looping over the keys in that map?

File: cmd/completion.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+`
+
+func MakeShellCompletion() *cobra.Command {
+
+	completion := &cobra.Command{
+		Use:   "completion SHELL",
+		Short: "Output shell completion for the given shell (bash or zsh)",
+		Long: `
+Outputs shell completion for the given shell (bash or zsh)
+This depends on the bash-completion binary.  Example installation instructions:
```
Comment (alexellis) [inline cmd/completion.go:26]: Can we download the bash-completion binary with arkade get?

File: cmd/get.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 
 // MakeGet creates the Get command to download software
 func MakeGet() *cobra.Command {
+	var validToolOptions []string = []string{"faas-cli", "helm", "inletsctl", "k3d", "k3sup", "kind", "kubectl", "kubectx", "kubeseal"}
```
Comment (alexellis) [inline cmd/get.go:25]: I don't think that this list is required, because you have it in `tools := get.MakeTools()`

File: cmd/completion.go (hunk 1 above)
Comment (tuananh) [reply cmd/completion.go:26]: @alexellis I suppose we can. Do you prefer it that way?

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**